realm/realm-library/src/main/cpp/io_realm_internal_objectstore_OsObjectBuilder.cpp

```diff
@@ -25,126 +25,126 @@ using namespace realm;
 using namespace realm::jni_util;
 using namespace realm::_impl;
 
-typedef std::vector<JavaValue> OsObjectData;
+typedef std::map<ColKey, JavaValue> OsObjectData;
```

Comment:
If you don't need the ordering, I think std::unordered_map is often faster.

Reply:
@finnschiermer we can't use `unordered_map` since the `std::hash` is not defined for `ColKey`, I know we can provide an implementation via template/`decltype` but this requires propagating this in other files as well. I don't think we'll gain much performance switching to an `unordered_map` since `std::map` is `O(log n)` and we use it to store a small list of properties/columns  (`std::map` is more memory efficient also :)